src/controller/CommissioningDelegate.h

```diff
@@ -77,38 +77,129 @@ class CommissioningParameters
     static constexpr size_t kMaxSsidLen          = 32;
     static constexpr size_t kMaxCredentialsLen   = 64;
 
+    // Value to use when setting the commissioning failsafe timer on the node being commissioned.
+    // If the failsafe timer value is passed in as part of the commissioning parameters, that value will be used. If not supplied,
+    // the AutoCommissioner will set this to the recommended value read from the node. If that is not set, it will fall back to the
+    // default kDefaultFailsafeTimeout.
+    // This value should be set before running PerformCommissioningStep for the kArmFailsafe step.
     const Optional<uint16_t> GetFailsafeTimerSeconds() const { return mFailsafeTimerSeconds; }
+
+    // The location (indoor/outdoor) of the node being commissioned.
+    // The node regulartory location (indoor/outdoor) should be set by the commissioner explicitly as it may be different than the
+    // location of the commissioner. This location will be set on the node if the node supports configurable regulatory location
+    // (from GetLocationCapability - see below). If the regulatory location is not supplied, this will fall back to the location in
+    // GetDefaultRegulatoryLocation and then to Outdoor (most restrictive).
+    // This value should be set before calling PerformCommissioningStep for the kConfigRegulatory step.
     const Optional<app::Clusters::GeneralCommissioning::RegulatoryLocationType> GetDeviceRegulatoryLocation() const
     {
         return mDeviceRegulatoryLocation;
     }
+
+    // Nonce sent to the node to use during the CSR request.
+    // When using the AutoCommissioner, this value will be ignored in favour of the value supplied by the
+    // OperationalCredentialsDelegate ObtainCsrNonce function. If the credential delegate is not supplied, the value supplied here
+    // will be used. If neither is supplied random value will be used as a fallback.
+    // This value must be set before calling PerformCommissioningStep for the kSendOpCertSigningRequest step.
     const Optional<ByteSpan> GetCSRNonce() const { return mCSRNonce; }
+
+    // Nonce value sent to the node to use during the attestation request.
+    // When using the AutoCommissioner, this value will fall back to random if not supplied.
+    // This value must be set before calling PerformCommissioningStep for the kSendAttestationRequest step.
```

Comment:
This is confusing.  This feels clearer to me:

```suggestion
    // Nonce value sent to the node to use during the attestation request.
    // When using the AutoCommissioner, this value will fall back to random if not supplied.
    // If a non-random value is to be used, the value must be set before calling PerformCommissioningStep for the kSendAttestationRequest step.
```